addons/purchase/stock.py

```diff
@@ -167,6 +167,49 @@ def _get_picking_to_recompute(self, cr, uid, ids, context=None):
                }),
     }
 
+    def action_view_purchase_order(self, cr, uid, ids, context=None):
+        pur_ids = set()
+        picking = self.browse(cr, uid, ids)[0]
+        result = self.pool['ir.actions.act_window'].for_xml_id(cr, uid, 'purchase', 'purchase_form_action', context=context)
+        for move in picking.move_lines.filtered(lambda x: x.purchase_line_id):
+            pur_ids.add(move.purchase_line_id.order_id.id)
+        if not pur_ids:
+            raise UserError(_('No Purchase Order Associated with this Incoming Shipment!'))
+        if len(list(pur_ids)) > 1:
+            result['domain'] = [('id', 'in', list(pur_ids))]
```

Comment:
same here. no need to check this condition . always posibility of one purchase order for incomming shipment.